### DISCUSSION TITLE
URQL - interface resolvers in graphcache

### DISCUSSION THREAD

**Discussion by 01jonathanf** [urql-graphql/urql#1624] [Q&A] [answered]
Hi all, needing some beginner help here,

I understand to relate a query to an entity in graphcache it can be done simply like so:

`getTodo: (_parent, args) => ({ __typename: "Todo", id: args.id })`

How can you create the same relationship with a query to an interface type which can return potentially multiple different typenames? e.g.: say I had an Animal interface and wanted the cache to lookup multiple typenames here:

`getAnimal: (_parent, args) => ({ __typename: could be "Dog", "Cat", etc..., id: args.id })`

**Comment by JoviDeCroock** [comment, marked as answer]
I think you'd be best off checking whether `cache.resolve({ __typename: 'Dog', id }, 'id') || cache.resolve({ __typename: 'cat', id }, 'id')` returns a field and use that. The `cache` is the thirth parameter of your function there.

**Reply by 01jonathanf** [reply]
Thanks, well currently what I've done to get it to work is this:

```
const { id } = args;
const typename =
              cache.resolve({ __typename: "Folder", id }, "__typename") ||
              cache.resolve({ __typename: "Journey", id }, "__typename");

return { __typename: typename, id: args.id };
```

so returning the typename from cache.resolve and then returning it in the object. is there a cleaner way to do this though directly from cache.resolve?

**Reply by JoviDeCroock** [reply]
Currently this seems like a good solution with our current API-surface
